libsolidity/analysis/ControlFlowBuilder.cpp

```diff
@@ -390,7 +390,7 @@ bool ControlFlowBuilder::visit(InlineAssembly const& _inlineAssembly)
 void ControlFlowBuilder::visit(yul::Statement const& _statement)
 {
 	solAssert(m_currentNode && m_inlineAssembly, "");
-	m_currentNode->location = langutil::SourceLocation::smallestCovering(m_currentNode->location, locationOf(_statement));
+	m_currentNode->location = langutil::SourceLocation::smallestCovering(m_currentNode->location, locationOf(_statement)->irLocation);
```

Comment:
Why isn't the `locationOf` helper just returning the location?

Reply:
I think the helper should just be fixed, and all these places using the location should use `locationOf` or `irLocationOf` (a new helper).